firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/FirebaseModelDownloader.java

```diff
@@ -187,7 +192,16 @@ public static FirebaseModelDownloader getInstance(@NonNull FirebaseApp app) {
    */
   @NonNull
   public Task<Void> deleteDownloadedModel(@NonNull String modelName) {
-    throw new UnsupportedOperationException("Not yet implemented.");
+
+    TaskCompletionSource<Void> taskCompletionSource = new TaskCompletionSource<>();
+    executor.execute(
+        () -> {
+          // remove all files associated with this model and then clean up model references.
+          fileManager.deleteAllModels(modelName);
```

Comment:
In case the app crashes, or the phone dies, after models have been deleted but before their details have been removed from sharedPreferences, what would happen?

Reply:
The model would be listed but when they tried to get the file it would return null... I think this is acceptable. I did update the getFile to handle this case a little more cleanly. If you can think of a better solution. I'm open to suggestions. I've also made a bug b/175144342 to add a start-up routine that will check that models with filePaths actually point to files and will clean them up if they don't.

Reply:
LGTM